### PR TITLE
Update interest rate

### PR DESCRIPTION
**Pull request by grandizzy**

### CODE REVIEW

File: src/ERC20Pool.sol [hunk 1]
```diff
@@ -287,10 +287,27 @@ contract ERC20Pool is IPool {
...
+        uint256 actualUtilization = getPoolActualUtilization();
+        require(
+            actualUtilization != 0 && previousRateUpdate < block.timestamp,
+            "ajna/interest-rate-not-updatable"
+        );
+
+        accumulatePoolInterest();
+        previousRate = Maths.wmul(
+            previousRate,
+            (Maths.sub(getPoolActualUtilization(), getPoolTargetUtilization()) +
```
Comment (EdNoepel) [inline src/ERC20Pool.sol]: Can we just reuse `actualUtilization` local variable here?

Comment (grandizzy) [reply src/ERC20Pool.sol]: good point, yeah, I was considering this but I think actual utilization will get changed after `accumulatePoolInterest `is called. Maybe first action should be to accumulate pool interest and then work with same `actualUtilization`?

Comment (grandizzy) [reply src/ERC20Pool.sol]: I pushed a change to change oreder of operations, pls have a look

File: src/ERC20Pool.sol [hunk 2]
```diff
@@ -287,10 +287,28 @@ contract ERC20Pool is IPool {
         emit Repay(msg.sender, lup, debtToPay);
     }
 
+    /// @notice Called by lenders to update interest rate of the pool when actual > target utilization
+    function updateInterestRate() external {
+        accumulatePoolInterest();
```
Comment (EdNoepel) [inline src/ERC20Pool.sol]: I'm uncomfortable with this:
- I'd rather the transaction quietly do nothing but burn a tiny bit of gas if the rate doesn't need to be updated.
- Though not strictly required, I'd rather the rate be calculated against the old utilization numbers prior to interest accumulation.

Comment (grandizzy) [reply src/ERC20Pool.sol]: please recheck

Comment (EdNoepel) [reply src/ERC20Pool.sol]: I'm still trying to rationalize the need for a revert.

For discussion purposes, why would this be undesirable?:
```
    function updateInterestRate() external {
        uint256 actualUtilization = getPoolActualUtilization();
        if (actualUtilization != 0 && previousRateUpdate < block.timestamp)
        {
            uint256 targetUtilization = getPoolTargetUtilization();
            
            accumulatePoolInterest();

            previousRate = Maths.wmul(
                previousRate,
                (Maths.sub(actualUtilization, targetUtilization) +
                    Maths.ONE_WAD)
            );
            previousRateUpdate = block.timestamp;
        }
    }
```

Comment (EdNoepel) [reply src/ERC20Pool.sol]: That would seemingly be more consistent with how `accumulatePoolInterest` behaves.

Comment (EdNoepel) [reply src/ERC20Pool.sol]: My motivation for avoiding a revert is that we want to incentivize updating the rate.  If we structure it such that it won't unwind a larger transaction, the proxy can bundle it with requests to withdraw a quote or draw debt.

Comment (grandizzy) [reply src/ERC20Pool.sol]: Ah, I understand now. This should be OK as long as we dont want to provide feedback to user calling the function. We can also expose a function like `isInterestRateUpdatable` that the user can call before calling `updateRateInterest`

Comment (grandizzy) [reply src/ERC20Pool.sol]: Pushed a new commit, pls check again

**Review by EdNoepel — APPROVED** [review]
👌

**Pull request merged**